osu.Game/Modes/UI/DrawableMod.cs

```diff
@@ -0,0 +1,97 @@
+﻿//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK.Graphics;
+using osu.Framework.Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
+
+namespace osu.Game.Modes.UI
+{
+    public class ModIcon : Container
+    {
+        private TextAwesome icon, bg;
+
+        private Color4 bgcolor;
+        public new Color4 Colour
+        {
+            get { return bgcolor; }
+            set { bgcolor = value; bg.Colour = value; }
+        }
+
+        private FontAwesome iconName;
+        public FontAwesome Icon
+        {
+            get { return iconName; }
+            set { iconName = value; icon.Icon = value; }
+        }
+
+        public ModIcon()
+        {
+            Children = new Drawable[]
+            {
+                bg = new TextAwesome
+                {
+                    Origin = Anchor.Centre,
+                    Anchor = Anchor.Centre,
+                    TextSize = 80,
+                    Icon = FontAwesome.fa_osu_mod_bg,
+                    Shadow = true,
+                },
+                icon = new TextAwesome
+                {
+                    Origin = Anchor.Centre,
+                    Anchor = Anchor.Centre,
+                    TextSize = 50,
+                    Colour = new Color4(84,84,84,255)
+                }
+            };
+        }
+    }
+    public class DrawableMod : Container
+    {
+        private ModIcon icon;
+        private SpriteText text;
+
+        private Color4 bgcolor;
+        public new Color4 Colour

```

Comment:
colour should probably be determined inside this class based on the mod, rather than passed in from an external source.